lib/queuePopulator/QueuePopulatorExtension.js

```diff
@@ -58,6 +58,8 @@ class QueuePopulatorExtension {
      * @return {undefined}
      */
     filter(entry) { // eslint-disable-line no-unused-vars
+        // sub classes can also implement 'filterAsync' for asynchronous
```

Comment:
I'd rather have `filter(entry)` default implementation do nothing then, it avoids to have to create a stub in the extension.

Reply:
Should I remove the assert here? also should I create a default `filterAsync` here?

Reply:
That looks ok to me, the assert was here to tell that `filter` was needed but now it's not the case anymore because we also support `filterAsync`.

Reply:
If you create a default `filterAsync` the logic you put in checking the existence of the function will not work anymore

Reply:
So, remove the assert?